scripts/composer/ScriptHandler.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * @file
+ * Contains \DrupalProject\composer\ScriptHandler.
+ */
+
+namespace DrupalProject\composer;
+
+use Composer\Script\Event;
+use Symfony\Component\Filesystem\Filesystem;
+
+class ScriptHandler {
+
+  protected static function getDrupalRoot($project_root) {
+    return $project_root .  '/web';
+  }
+
+  public static function buildScaffold(Event $event) {
+    $fs = new Filesystem();
+    if (!$fs->exists(static::getDrupalRoot(getcwd()) . '/autoload.php')) {
+      \DrupalComposer\DrupalScaffold\Plugin::scaffold($event);
+    }
+  }
+
+  public static function createRequiredFiles(Event $event) {
+    $fs = new Filesystem();
+    $root = static::getDrupalRoot(getcwd());
+
+    $dirs = [
+      'modules',
+      'profiles',
+      'themes',
+    ];
+
+    // Required for unit testing
+    foreach ($dirs as $dir) {
+      if (!$fs->exists($root . '/'. $dir)) {
+        $fs->mkdir($root . '/'. $dir);
+        $fs->touch($root . '/'. $dir . '/.gitkeep');
+      }
+    }
+
+    // Prepare the settings file for installation
```

Comment:
I think we should allow user to turn copying file on or off using some variable in `composer.json`. If you like the idea we can create a new issue for this.